astropy/modeling/mappings.py

```diff
@@ -34,8 +38,8 @@ def mapping(self):
         return self._mapping
 
     def evaluate(self, *args):
-        if len(args) < self.n_inputs:
-            raise TypeError('{0} expects at most {1} inputs; got {2}'.format(
+        if len(args) != self.n_inputs:
+            raise TypeError('{0} expects {1} inputs; got {2}'.format(
```

Comment:
Note that in this check `self.n_inputs` is equal to `max(mapping)` if `n_inputs = None`.

With that we have 3 cases:

- swapping axes: all indices must be specified in `mapping`, in order to get the correct order, so strict checking is OK in this case.
- dropping axes: `n_inputs` must be specified in order to correctly chain this with other models - strict checking is OK.
- duplicating axes: all axes' indices must be in `mapping` in this case as well, and `self.n_inputs=max(self.mapping)`, so strict checking is OK in this case as well.

Am I missing something?